src/core/dependency-injection/container-init.ts

```diff
@@ -67,138 +76,98 @@ export class Container {
    * @param logLevel - the log level to use, defaults to 'debug'
    * @param developmentMode - if true, show full stack traces in error messages
    * @param testLogger - a test logger to use, if provided
+   * @param overrides - mocked instances to use instead of the default implementations
```

Comment:
```suggestion
   * @param overrides - instances to use instead of the default implementations
```
we should never change implementation just for testing purposes.  I think the implementation is fine, but we shouldn't mention that it is for mocking, as it could be used for other purposes as well.